src/test/groovy/graphql/execution/reactive/CompletionStageOrderedSubscriberTest.groovy

```diff
@@ -0,0 +1,201 @@
+package graphql.execution.reactive
+
+import graphql.execution.pubsub.CapturingSubscriber
+import graphql.execution.pubsub.CapturingSubscription
+import spock.lang.Specification
+
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionStage
+import java.util.function.Function
+
+import static graphql.execution.reactive.CompletionStageSubscriberTest.mapperThatDoesNotComplete
+
+class CompletionStageOrderedSubscriberTest extends Specification {
```

Comment:
by breaking out the Subscriber classes into their own class - we can unit test them better